src/DotLiquid.Tests/StandardFilterTests.cs

```diff
@@ -195,6 +195,24 @@ public void TestSlice()
             Assert.AreEqual("efg", StandardFilters.Slice("abcdefg", 4, 30));
             Assert.AreEqual("a", StandardFilters.Slice("abc", -4, 2));
             Assert.AreEqual("", StandardFilters.Slice("abcdefg", -10, 1));
+
+            // Test for arrays
+            var testArray = new string[] { "a", "b", "c", "d" };
+            Assert.AreEqual("a", StandardFilters.Slice(testArray, 0));
+            Assert.AreEqual("bc", StandardFilters.Slice(testArray, 1, 2));
+            Assert.AreEqual("c", StandardFilters.Slice(testArray, -2, 1));
+            Assert.AreEqual("d", StandardFilters.Slice(testArray, -1, 1));

```

Comment:
@microalps, I've replicated the Ruby tests, which highlighted a small difference between the implementations.
DotLiquid returns null for an empty slice of a string input, whereas Liquid returns and empty string.

I've opted to go with the Liquid approach and return an empty string as this feels like a bug.
If you think this should be changed back so that DotLiquid return a null let me know.

Reply:
Reinstated DotLiquid logic by default and introduced Liquid fix behind compatibility flag 2.2a